src/thunderbots/software/xbox_controller_mapping/main.cpp

```diff
@@ -0,0 +1,176 @@
+    // http://wiki.ros.org/joy section 5.3
+    enum ControllerAxes_t
+    {
+        LEFTRIGHT_AXIS_STICK_LEFT = 0,
+        UPDOWN_AXIS_STICK_LEFT,
+        LEFT_TRIGGER,
+        LEFTRIGHT_AXIS_STICK_RIGHT,
+        UPDOWN_AXIS_STICK_RIGHT,
+        RIGHT_TRIGGER,
+        CROSS_PAD_LEFT_RIGHT,
+        CROSS_PAD_UP_DOWN
+    } ControllerAxes;
+}  // namespace
+
+/**
+ * Toggles whether or not the dribbler is running every time the given button is pressed
+ * (value of 1 for pressed, 0 for not pressed). Acts like a D Flip-Flop.
+ *
+ * @param button_pressed The value indicating if the button is pressed. 1 indicates the
+ * button is pressed, 0 indicates the button is not pressed
+ *
+ * @return 1 if the dribbler should be enabled, and 0 if the dribbler should not be
+ * enabled
+ */
+unsigned int toggleDribble(unsigned int button_pressed)
+{
+    static unsigned int dribble_enabled  = 0;
+    static bool button_has_been_released = true;
+
+    if (button_pressed == 1 && button_has_been_released)
+    {
+        dribble_enabled          = dribble_enabled == 1 ? 0 : 1;
+        button_has_been_released = false;
+    }
+    else if (button_pressed == 0 && !button_has_been_released)
+    {
+        button_has_been_released = true;
+    }
+
+    return dribble_enabled;
+}
+
+void joystickUpdateCallback(const sensor_msgs::Joy::ConstPtr &msg)
+{
+    sensor_msgs::Joy::_axes_type axes       = msg->axes;
+    sensor_msgs::Joy::_buttons_type buttons = msg->buttons;
+
+    // Axes values report values in the range [-1.0, 1.0]. Values are 1.0 when the
+    // joystick is pushed all the way up, -1.0 when the joystick is all the way down,
+    // 1.0 when the joystick is pushed all the way left, and -1.0 when the joystick is
+    // pushed all the way right
+    double robot_x_velocity =
+        axes[UPDOWN_AXIS_STICK_LEFT] * ROBOT_MAX_SPEED_METERS_PER_SECOND *
+        Util::DynamicParameters::XBoxControllerDemo::linear_sensitivity.value();
+    double robot_y_velocity =
+        axes[LEFTRIGHT_AXIS_STICK_LEFT] * ROBOT_MAX_SPEED_METERS_PER_SECOND *
+        Util::DynamicParameters::XBoxControllerDemo::linear_sensitivity.value();
+    double robot_angular_velocity =
+        axes[LEFTRIGHT_AXIS_STICK_RIGHT] * ROBOT_MAX_ANG_SPEED_RAD_PER_SECOND *
+        Util::DynamicParameters::XBoxControllerDemo::angular_sensitivity.value();
+
+    // The Y button reports an integer of 1 when pressed, and 0 when not pressed
+    double robot_dribbler_rpm =
+        toggleDribble(static_cast<unsigned int>(buttons[Y])) *
+        Util::DynamicParameters::XBoxControllerDemo::dribbler_rpm.value();
+
+    // The triggers report values in the range [-1.0, 1.0]. Values are < 0 when the
+    // trigger is pressed/held down more than halfway, and > 0 otherwise
+    double robot_kick_speed_meters_per_second =
+        (axes[RIGHT_TRIGGER] < 0.0)
+            ? Util::DynamicParameters::XBoxControllerDemo::kick_speed_meters_per_second
+                  .value()
+            : 0.0;
```

Comment:
Could we change this to something like:
```
   double robot_kick_speed_meters_per_second = Util::DynamicParameters::XBoxControllerDemo::kick_speed_meters_per_second.value();
   if (axes[RIGHT_TRIGGER] < 0.0) {
        robot_kick_speed_meters_per_second = 0;
   }
```
and similarly below?

Reply:
fixed

Reply:
Doesn't look like this has changed?

Reply:
Sorry, pushed now